app/src/test/java/com/shagalalab/constitution/ui/article/ArticleViewModelTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.shagalalab.constitution.ui.article
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.shagalalab.constitution.TestExecutor
+import com.shagalalab.constitution.data.dao.ArticleDao
+import com.shagalalab.constitution.data.models.ArticleModel
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class ArticleViewModelTest {
+
+    private lateinit var viewModel: ArticleViewModel
+    private val articleModel = ArticleModel(1)
+    private val articleModels = listOf(articleModel)
+    private val executor = TestExecutor()
+    private val articleMockDao = mockk<ArticleDao>()
+
+    @Before
+    fun setUp() {
+        viewModel = ArticleViewModel(articleMockDao, executor)
+    }
+
+    @get: Rule
+    val rule = InstantTaskExecutorRule()
+
+    @Test
+    fun getArticlesByChapterId() {
+        every { articleMockDao.getArticlesByChapterId(1) } returns articleModels
+        viewModel.getArticlesByChapterId(1)
+        val result = viewModel.articleList.value
+        Assert.assertEquals(result, articleModels)
+    }
+
+    @Test
+    fun getArticlesByPartId() {
+        every { articleMockDao.getArticlesByPartId(1) } returns articleModels
```

Comment:
Do you remember, I mentioned about three phases of individual test:
- given
- when
- then

Let's divide to those phases by adding the section names as comment line. If you recall the sample test that I provided, I put those sections there.